File: solution_set1/1679. Max Number of K-Sum Pairs.py
##ss
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        
        pairs = 0
        seen = {}
        for x in range(len(nums)):
            if k - nums[x] in seen:
                seen[k-nums[x]]-=1
                pairs+=1
                
                if seen[k-nums[x]] == 0:
                    seen.pop(k-nums[x])
                    
            else:
                seen[nums[x]] = seen.get(nums[x],0) + 1
                
                
        return pairs
        

File: cpp_solutions/1679. Max Number of K-Sum Pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        unordered_map<int,int> seen;
        
        
        int c = 0;
        
        for(int x=0; x<nums.size();x++){
            
            int key = k - nums[x];
            
            if (seen.find(key)!=seen.end() and seen[key]!=0){
                c+=1;
                seen[key]-=1;
                //seen.erase(key);
            }
            else{
                if(seen.find(nums[x])!=seen.end()){
                    seen[nums[x]]+=1;
                }
                else{
                    seen[nums[x]] = 1;
                }
                //seen.insert(nums[x]);
            }
        }
        return c;
        
    }
};
